syntax = "proto3";

package MolecularPropertyProfile;

message FloatValueCount {
  float value = 1;
  int32 count = 2;
}

message IntValueCount {
  int32 value = 1;
  int32 count = 2;
}

message Quantile {
  float quantile = 1;
  float value = 2;
}

message Description {
  // The name of the feature, column name in the input.
  string feature_name = 1;
  // A human appropriate description of this feature.
  string description = 2;
  // What to show when this is plotted.
  string plot_title = 3;
  // The colour any plot using this data.
  string line_color = 4;
  // The source of this data - Chembl, Pubchem, ...
  string source = 5;
}

message Descriptor {
  Description description = 1;

  // The number of samples used to generate this profile.
  int32 n = 3;
  // Simple statistics.
  // Uses float even though the feature itself may only have integral values
  float minval = 4;
  float maxval = 5;
  float mean = 6;
  float median = 7;
  repeated Quantile quantile = 8;

  // Depending on the kind of data, only one of these will be populated.
  repeated IntValueCount int_values = 9;
  repeated FloatValueCount float_values = 10;
}
